src/PSRule.Rules.Azure/rules/Azure.VNET.Rule.ps1

```diff
@@ -227,6 +227,22 @@ Rule 'Azure.LB.Name' -Type 'Microsoft.Network/loadBalancers' -Tag @{ release = '
     Match 'Name' '^[A-Za-z0-9]((-|\.)*\w){0,79}$'
 }
 
+# Synopsis: Load balancers deployed with Standard SKU should be zone-redundant for high availability.
+Rule 'Azure.LB.AvailabilityZone' -Type 'Microsoft.Network/loadBalancers' -If { IsStandardLoadBalancer } -Tag @{ release = 'GA'; ruleSet = '2021_09'; } {
+    foreach ($ipConfig in $TargetObject.Properties.frontendIPConfigurations) {
+        $zonesNotSet = $Assert.NullOrEmpty($ipConfig, 'zones').Result;
+
+        $zoneRedundant = -not ($ipConfig.zones -and (Compare-Object -ReferenceObject @('1','2','3') -DifferenceObject $ipConfig.zones));
```

Comment:
Might be a nicer way to do this. Wanted to use `Compare-Object` so we don't care about order.

Reply:
I did have a think about this one and the closest would be:

```powershell
$Assert.HasFieldValue($ipConfig, 'zones.Length', 3);
$Assert.In($ipConfig, 'zones', @('1', '2', '3'));
```

But it's not quite the same since it would allow `'1', '1', '1'`.